litestar/datastructures/headers.py

```diff
@@ -235,7 +237,8 @@ def __iter__(self) -> Iterator[str]:
         return iter(h[0].decode("latin-1") for h in self.headers)
 
 
-class Header(BaseModel, ABC):
+@dataclass(kw_only=True)
```

Comment:
```suggestion
@dataclass
```
`kw_only` parameter only exists in python 3.10+ - we need to support 3.8+